spec/unit/rest/logging_spec.rb

```diff
@@ -0,0 +1,74 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require 'gooddata/rest/rest'
+
+require_relative '../../../lib/gooddata/bricks/middleware/logger_middleware'
+
+describe 'Behavior during api calls' do
+  before :each do
+    WebMock.enable!
+
+    @server = GoodData::Environment::ConnectionHelper::DEFAULT_SERVER
+    @coef = GoodData::Rest::Connection::RETRY_TIME_COEFFICIENT
+    @init = GoodData::Rest::Connection::RETRY_TIME_INITIAL_VALUE
+    @poll = GoodData::Rest::Client::DEFAULT_SLEEP_INTERVAL
+
+    GoodData::Rest::Connection.send(:remove_const, :RETRY_TIME_COEFFICIENT) if GoodData::Rest::Connection.const_defined?(:RETRY_TIME_COEFFICIENT)
+    GoodData::Rest::Connection.send(:remove_const, :RETRY_TIME_INITIAL_VALUE) if GoodData::Rest::Connection.const_defined?(:RETRY_TIME_INITIAL_VALUE)
+    GoodData::Rest::Client.send(:remove_const, :DEFAULT_SLEEP_INTERVAL) if GoodData::Rest::Client.const_defined?(:DEFAULT_SLEEP_INTERVAL)
+
+    GoodData::Rest::Connection.const_set(:RETRY_TIME_COEFFICIENT, 1)
+    GoodData::Rest::Connection.const_set(:RETRY_TIME_INITIAL_VALUE, 0)
+    GoodData::Rest::Client.const_set(:DEFAULT_SLEEP_INTERVAL, 0)
+    stub_request(:get, "#{@server}/gdc")
+    .to_return(
+      :body => {
+        'about' => {
+          'links' => [{
+            'link' => 'https://secure-di.gooddata.com/uploads',
+            'summary' => 'User data staging area.',
+            'category' => 'uploads',
+            'title' => 'user-uploads'
+          }]
+        }
+      }.to_json,
+      :headers => { 'Content-Type' => 'application/json' }
+    )
+    stub_request(:post, "#{@server}/gdc/account/login")
+      .to_return(:body => { :userLogin => { :profile => "/profile/123" } }.to_json, :headers => { 'Content-Type' => "application/json" })
+    stub_request(:get, "#{@server}/gdc/account/token")
+      .to_return(:body => {}.to_json, :headers => { 'Content-Type' => "application/json" })
+    stub_request(:get, "#{@server}/profile/123")
+      .to_return(:body => {}.to_json, :headers => { 'Content-Type' => "application/json" })
+    stub_request(:get, "#{@server}/poll_test")
+      .to_return(:body => { stuff: :aaa }.to_json, :headers => { 'Content-Type' => "application/json" })
+    stub_request(:get, "#{@server}/too_many_reqs")
```

Comment:
Similar code found in logging_spec.rb:49, logging_spec.rb:51, logging_spec.rb:53 (mass = 51)